core/string/ustring.cpp

```diff
@@ -4699,11 +4699,16 @@ String String::property_name_encode() const {
 
 static const char32_t invalid_node_name_characters[] = { '.', ':', '@', '/', '\"', UNIQUE_NODE_PREFIX[0], 0 };
 
-String String::get_invalid_node_name_characters() {
+String String::get_invalid_node_name_characters(bool p_allow_internal) {
 	// Do not use this function for critical validation.
 	String r;
 	const char32_t *c = invalid_node_name_characters;
 	while (*c) {
+		if (p_allow_internal && String::chr(*c) == "@") {
```

Comment:
Wouldn't this work?
```suggestion
		if (p_allow_internal && *c == '@') {
```

Reply:
I'm not sure, this is a bit too low level for me to grasp intuitively. This definitely worked, but I wasn't confident if doing it is safe:

```suggestion
		if (p_allow_internal && c[0] == '@') {
```

Reply:
I think it's safe, but not sure how it handles unicode.

Reply:
CC @bruvzg

Reply:
I've tested KoBeWi's suggestion, and it works. It seems to match what `String::validate_node_name()` does too, so I'll push that version.